### PR TITLE
Fed identity for key vault

### PR DESCRIPTION
**Pull request by satr**

### CODE REVIEW

File: src/components/component/component-identity.tsx [hunk 1]
```diff
@@ -59,12 +58,27 @@ const AzureIdentityLink = ({
...
+              <Typography className="whitespace-nowrap" variant="h6" as="span">
+                Azure Key Vaults using Azure identity
+              </Typography>
+              <List variant="bullet">
+                {azureKeyVaults.map((name: string) => (
+                  <List.Item key={name}>{name}</List.Item>
+                ))}
+              </List>
+            </div>
+          )}
```
Comment (nilsgstrabo) [inline src/components/component/component-identity.tsx:81]:
```suggestion
          <div className="grid grid--gap-medium">
            <AzureIdentity
              oidcIssuerUrl={configVariables.CLUSTER_OIDC_ISSUER_URL}
              clientId={clientId}
              namespace={namespace}
              serviceAccountName={serviceAccountName}
            />
            {azureKeyVaults?.length > 0 && (
              <div className="grid grid--gap-small">
                <Typography
                  className="whitespace-nowrap"
                  variant="h6"
                  as="span"
                >
                  Azure Key Vaults using Azure identity
                </Typography>
                <List variant="bullet">
                  {azureKeyVaults.map((name: string) => (
                    <List.Item key={name}>{name}</List.Item>
                  ))}
                </List>
              </div>
            )}
          </div>
```
Suggest we wrap all elements within Popover.Content inside a grid with medium gap, and then Azure KeyVault info inside a grid with small gap.

Also removed unused azureKeyVaults prop for AzureIdentity component.

File: src/components/identity/azure-identity.tsx [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { Icon, Typography } from '@equinor/eds-core-react';
+import { Accordion, Icon, Typography } from '@equinor/eds-core-react';
```
Comment (nilsgstrabo) [inline src/components/identity/azure-identity.tsx:1]:
```suggestion
import { Icon, Typography } from '@equinor/eds-core-react';
```
Remove unused import

File: src/components/identity/azure-identity.tsx [hunk 3]
```diff
@@ -30,13 +30,15 @@ export interface AzureIdentityProps {
   namespace: string;
   serviceAccountName: string;
   clientId: string;
+  azureKeyVaults?: Array<string>;
```
Comment (nilsgstrabo) [inline src/components/identity/azure-identity.tsx:33]: Not used, can be removed

File: src/components/identity/azure-identity.tsx [hunk 4]
```diff
@@ -30,13 +30,15 @@ export interface AzureIdentityProps {
...
 }
 
 export const AzureIdentity = ({
   oidcIssuerUrl,
   namespace,
   serviceAccountName,
   clientId,
+  azureKeyVaults,
```
Comment (nilsgstrabo) [inline src/components/identity/azure-identity.tsx:41]: Not used, can be removed

File: src/components/identity/azure-identity.tsx [hunk 5]
```diff
@@ -76,4 +78,5 @@ AzureIdentity.propTypes = {
   namespace: PropTypes.string.isRequired,
   serviceAccountName: PropTypes.string.isRequired,
   clientId: PropTypes.string.isRequired,
+  azureKeyVaults: PropTypes.arrayOf(PropTypes.string),
```
Comment (nilsgstrabo) [inline src/components/identity/azure-identity.tsx:81]: Not used, can be removed

**Review by nilsgstrabo — CHANGES_REQUESTED** [review]
Added some suggestions

File: src/components/component/secrets/secret-list-item.tsx [hunk 6]
```diff
@@ -28,7 +28,7 @@ export const SecretListItem = ({
 }): JSX.Element => {
   const [isExpanded, setIsExpanded] = useState(false);
 
-  const hasStatusMessages = secret.statusMessages?.length > 0;
+  const hasStatusMessages = secret?.statusMessages?.length > 0;
```
[the remark this answers is not in the document]
Comment (satr) [reply src/components/component/secrets/secret-list-item.tsx:31]: Removed.

File: src/components/identity/azure-identity.tsx (hunk 3 above)
Comment (satr) [reply src/components/identity/azure-identity.tsx:33]: removed

File: src/components/identity/azure-identity.tsx (hunk 4 above)
Comment (satr) [reply src/components/identity/azure-identity.tsx:41]: removed

File: src/components/identity/azure-identity.tsx (hunk 5 above)
Comment (satr) [reply src/components/identity/azure-identity.tsx:81]: removed

**Review by nilsgstrabo — APPROVED** [review]

**Pull request merged**